tools/restatectl/src/commands/log/reconfigure.rs

```diff
@@ -0,0 +1,165 @@
+    let channel = grpc_connect(connection.cluster_controller.clone())
+        .await
+        .with_context(|| {
+            format!(
+                "cannot connect to cluster controller at {}",
+                connection.cluster_controller
+            )
+        })?;
+
+    let mut client =
+        ClusterCtrlSvcClient::new(channel).accept_compressed(CompressionEncoding::Gzip);
+
+    let (provider, params) = match opts.provider {
+        ProviderKind::Local => (ProtoProviderKind::Local, rand::random::<u64>().to_string()),
+        #[cfg(any(test, feature = "memory-loglet"))]
+        ProviderKind::InMemory => (ProtoProviderKind::InMemory, String::default()),
+        #[cfg(feature = "replicated-loglet")]
+        ProviderKind::Replicated => (
+            ProtoProviderKind::Replicated,
+            replicated_loglet_params(&mut client, opts).await?,
+        ),
+        #[allow(unreachable_patterns)]
+        _ => anyhow::bail!("Provider kind is not supported"),
+    };
+
+    client
+        .seal_and_extend_chain(SealAndExtendChainRequest {
+            log_id: opts.log_id,
+            min_version: Some(Version {
+                value: opts.min_version.get(),
+            }),
+            provider: provider.into(),
+            segment_index: opts.segment_index,
+            params,
+        })
+        .await?;
+
+    Ok(())
+}
+
+async fn replicated_loglet_params(
+    client: &mut ClusterCtrlSvcClient<Channel>,
+    opts: &ReconfigureOpts,
+) -> anyhow::Result<String> {
+    let mut logs_resposne = client
+        .list_logs(ListLogsRequest {})
+        .await
+        .context("Failed to get logs metadata")?
+        .into_inner();
+
+    let logs = StorageCodec::decode::<Logs, _>(&mut logs_resposne.logs)?;
+    let chain = logs
+        .chain(&LogId::from(opts.log_id))
+        .with_context(|| format!("Unknown log id '{}'", opts.log_id))?;
+
+    let tail_index = opts.segment_index.unwrap_or(chain.tail_index().into());
+
+    // format is, log_id in the higher order u32, and segment_index in the lower
+    let loglet_id: u64 =
+        (u64::from(opts.log_id) << (size_of::<LogId>() * 8)) + u64::from(tail_index + 1);
+
+    let tail_segment = chain.tail();
+
+    let params = if tail_segment.config.kind == ProviderKind::Replicated {
+        let last_params =
+            ReplicatedLogletParams::deserialize_from(tail_segment.config.params.as_bytes())
+                .context("Last segment params in chain is invalid")?;
+
+        ReplicatedLogletParams {
+            loglet_id: loglet_id.into(),
+            nodeset: if opts.nodeset.is_empty() {
+                last_params.nodeset
+            } else {
+                NodeSet::from_iter(opts.nodeset.iter().cloned())
+            },
+            replication: ReplicationProperty::new(opts.replication_factor.unwrap_or_else(|| {
+                NonZeroU8::new(last_params.replication.num_copies())
```

Comment:
What if the previous segment has location-aware replication property? shouldn't we just copy it as is instead of flattening it?